src/pages/events-description/index.js

```diff
@@ -24,25 +25,66 @@ export const EventsPageDescription = () => {
   const [event, setEvent] = useState([])
   const [thumbsSwiper, setThumbsSwiper] = useState(null)
   const [galeria, setGaleria] = useState([])
+  const [activities, setActivities] = useState([])
   const { id } = useParams()
+  const activitiesList = []
   useEffect(() => {
-    cms.get(`api/events/${id}/?populate=parceires&populate=fotos_evento`).then((response) => {
-      const { data } = response.data
-      setEvent(data)
+    cms
+      .get(
+        `api/events/${id}/?populate=parceires
+        &populate=fotos_evento
+        &populate=subeventos
+        &populate=oficinas
+        &populate=oficinas.foto_divulgacao
+        &populate=subeventos.foto_divulgacao`
+      )
+      .then((response) => {
+        const { data } = response.data
+        setEvent(data)
 
-      const fotoDivulgacao = data.attributes.fotos_evento
-      const images = fotoDivulgacao.data.map(
-        (image) => {
+        const fotoDivulgacao = data.attributes.fotos_evento
+        const images = fotoDivulgacao.data.map((image) => {
           return {
             id: image.id,
             name: image.attributes?.name,
             url: process.env.REACT_APP_URL_CMS + image.attributes?.url
           }
-        }
-      )
-      setGaleria(images)
-    })
-  }, [])
+        })
+        setGaleria(images)
+        const oficinas = data.attributes.oficinas.data.map((oficina) => {
+          console.log(oficina)
+          return {
+            id: oficina.id,
+            type: 'workshops',
+            name: oficina.attributes.nome,
+            url:
+            process.env.REACT_APP_URL_CMS +
+              oficina.attributes.foto_divulgacao.data[0].attributes.url,
+            data_inicio: oficina.attributes.data_inicio,
+            horario_inicio: oficina.attributes.horario_inicio
+          }
+        })
+        activitiesList.push(oficinas)
+        console.log('oficinas', data.attributes.oficinas.data)
+        console.log('subeventos', data.attributes.subeventos.data)
+        const subeventos = data.attributes.subeventos.data.map((evento) => {
+          return {
+            type: 'events',
+            id: evento.id,
+            name: evento.attributes.nome,
+            url:
+            process.env.REACT_APP_URL_CMS +
+              evento.attributes.foto_divulgacao.data.attributes.url,
+            data_inicio: evento.attributes.data_inicio,
+            horario_inicio: evento.attributes.horario_inicio
+          }
+          // })
+        })
+        activitiesList.push(subeventos)
+        setActivities(activitiesList)
```

Comment:
Isso aqui está um pouco estranho, não sei se deveríamos alterar o estado manualmente (sem usar setActivities). O React preza por imutabilidade então talvez isso tenha consequências inesperadas. Minha sugestão é usar destructuring ao invés de push:

```suggestion
        activitiesList.push(subeventos)
        setActivities([
          ...activitiesList
          ...subeventos
        ])
```

Nesse caso, o destructuring cria uma lista nova ao invés de alterar a lista existente, mantendo o comportamento de imutabilidade esperado pelo React.

Reply:
Vi a versão final agora, tá perfeito! Desde que a gente não altere `activities` diretamente como estávamos fazendo com push antes.

Entretanto, do jeito que está, estamos criando um array de arrays, certo?

Um array contendo dois arrays, mais precisamente, o primeiro com `subeventos` e o segundo com `oficinas`. Não sei se isso é o que vocês queriam fazer mesmo. Se não for, dá para usar concat para deixar tudo num array único. Seria algo mais ou menos assim:

```
  setActivities(subeventos.concat(oficinas))
```

Reply:
Lendo o código agora, estamos fazendo map dentro de map, provavelmente porque temos esse array de arrays, eu acho que o mais eficiente seria ter apenas um array com todos os items de subeventos e oficinas, dado que um subevento e uma oficina têm os mesmos atributos, então é ok eles ficarem num mesmo array.

Acho que seria legal olharmos isso juntos no Discord, essa mudança é um pouco confusa.